ext/http/00_serve.js

```diff
@@ -0,0 +1,466 @@
+    }
+    return {
+      transport: "tcp",
+      hostname: this.#methodAndUri[3],
+      port: this.#methodAndUri[4],
+    };
+  }
+
+  get method() {
+    if (this.#methodAndUri === undefined) {
+      if (this.#slabId === undefined) {
+        throw new TypeError("request closed");
+      }
+      this.#methodAndUri = core.ops.op_get_request_method_and_url(this.#slabId);
+    }
+    return this.#methodAndUri[0];
+  }
+
+  get body() {
+    if (this.#slabId === undefined) {
+      throw new TypeError("request closed");
+    }
+    if (this.#body !== undefined) {
+      return this.#body;
+    }
+    // If the method is GET or HEAD, we do not want to include a body here, even if the Rust
+    // side of the code is willing to provide it to us.
+    if (this.method == "GET" || this.method == "HEAD") {
+      this.#body = null;
+      return null;
+    }
+    this.#streamRid = core.ops.op_read_request_body(this.#slabId);
+    this.#body = new InnerBody(readableStreamForRid(this.#streamRid, false));
+    return this.#body;
+  }
+
+  get headerList() {
+    if (this.#slabId === undefined) {
+      throw new TypeError("request closed");
+    }
+    return core.ops.op_get_request_headers(this.#slabId);
+  }
+
+  get headers() {
+    if (!this.#headers) {
+      if (this.#slabId === undefined) {
+        throw new TypeError("request closed");
+      }
+      this.#headers = new RequestHeaders(this.#slabId);
+    }
+    return this.#headers;
+  }
+
+  get slabId() {
+    return this.#slabId;
+  }
+}
+
+class CallbackContext {
+  scheme;
+  fallbackHost;
+  serverRid;
+
+  initialize(scheme, fallbackHost) {
+    this.scheme = scheme;
+    this.fallbackHost = fallbackHost;
+  }
+}
+
+function fastSyncResponseOrStream(req, respBody) {
+  if (respBody === null || respBody === undefined) {
+    // Don't set the body
+    return null;
+  }
+
+  const stream = respBody.streamOrStatic;
+  const body = stream.body;
+
+  if (ObjectPrototypeIsPrototypeOf(Uint8ArrayPrototype, body)) {
+    core.ops.op_set_response_body_bytes(req, body);
+    return null;
+  }
+
+  if (typeof body === "string") {
+    core.ops.op_set_response_body_text(req, body);
+    return null;
+  }
+
+  // At this point in the response it needs to be a stream
+  if (!ObjectPrototypeIsPrototypeOf(ReadableStreamPrototype, stream)) {
+    throw TypeError("invalid response");
+  }
+  const resourceBacking = getReadableStreamResourceBacking(stream);
+  if (resourceBacking) {
+    core.ops.op_set_response_body_resource(
+      req,
+      resourceBacking.rid,
+      resourceBacking.autoClose,
+    );
+    return null;
+  }
+
+  return stream;
+}
+
+async function asyncResponse(responseBodies, req, status, stream) {
+  const responseRid = core.ops.op_set_response_body_stream(req);
+  SetPrototypeAdd(responseBodies, responseRid);
+  const reader = stream.getReader();
+  const responseStream = writableStreamForRid(responseRid, true);
```

Comment:
This is not good. There is a lot of overhead in creating and using a `WritableStream` (especially garbage). Use `core.writeAll(rid, bytes)` directly instead.

Reply:
This code was adapted from existing code 01_http.js (https://github.com/denoland/deno/blob/main/ext/http/01_http.js#L318) but using `core.writeAll` sounds like a better fit regardless.

Reply:
Fixed.